task/build-paketo-builder-oci-ta/0.1/build-paketo-builder-oci-ta.yaml

```diff
@@ -185,17 +191,43 @@ spec:
 
         echo "Podman version"
         podman version
-
-        echo "Podman info"
-        podman info
         REMOTESSHEOF
         chmod +x scripts/script-setup.sh
 
         cat >scripts/script-build.sh <<'REMOTESSHEOF'
         #!/bin/sh
-
         cd /var/workdir
 
+        echo "Fetch the dockerfile using param: $DOCKERFILE"
+        if [ -e "$SOURCE_CODE_DIR/$CONTEXT/$DOCKERFILE" ]; then
+          dockerfile_path="$(pwd)/$SOURCE_CODE_DIR/$CONTEXT/$DOCKERFILE"
+        elif [ -e "$SOURCE_CODE_DIR/$DOCKERFILE" ]; then
+          dockerfile_path="$(pwd)/$SOURCE_CODE_DIR/$DOCKERFILE"
+        elif [ -e "$DOCKERFILE" ]; then
+          # Custom Dockerfile location is mainly used for instrumented builds for SAST scanning and analyzing.
+          # Instrumented builds use this step as their base and also need to provide modified Dockerfile.
+          dockerfile_path="$DOCKERFILE"
+        elif echo "$DOCKERFILE" | grep -q "^https\?://"; then
+          echo "Fetch Dockerfile from $DOCKERFILE"
+          dockerfile_path=$(mktemp --suffix=-Dockerfile)
+          http_code=$(curl -s -S -L -w "%{http_code}" --output "$dockerfile_path" "$DOCKERFILE")
+          if [ $http_code != 200 ]; then
+            echo "No Dockerfile is fetched. Server responds $http_code"
+            exit 1
+          fi
+          http_code=$(curl -s -S -L -w "%{http_code}" --output "$dockerfile_path.dockerignore.tmp" "$DOCKERFILE.dockerignore")
+          if [ $http_code = 200 ]; then
+            echo "Fetched .dockerignore from $DOCKERFILE.dockerignore"
+            mv "$dockerfile_path.dockerignore.tmp" $SOURCE_CODE_DIR/$CONTEXT/.dockerignore
+          fi
+        else
+          echo "Cannot find Dockerfile $DOCKERFILE"
+          exit 1
+        fi
```

Comment:
Although it is not a problem introduced by this pull request, I do not like how the code to discover `Dockerfile` is copy-pasted on so many places.  I have a very similar block of code in #1653 and I already asked about this in the related [Jira issue](https://issues.redhat.com/browse/STONEBLD-2804?focusedId=25945741&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25945741).  Do we have a separate Jira issue for this?

Reply:
> Do we have a separate Jira issue for this?

No.

Reply:
I have created https://issues.redhat.com/browse/KONFLUX-6400 for this.

Reply:
See fixes: 6e766ca..ed97dc3

Reply:
Is it ok with my changes ? @kdudka

Reply:
@cmoulliard I like how the duplicated code was removed.  I do not feel competent to comment on the other changes though.